### DISCUSSION TITLE
rake generate_derivatives issue with Ruby 3.2+ ("NoMethodError: undefined method `exists?' for Dir:Class")

### DISCUSSION THREAD

**Discussion by dcnb** [CollectionBuilder/collectionbuilder.github.io#94] [General]
Hi everyone, 

if you're trying to use the rake task "generate_derivatives" to create derivative objects from your objects folder, you may run into an error message like this: 

`NoMethodError: undefined method `exists?' for Dir:Class`

The ".exists" command no longer works in Ruby 3.2 and higher. So if you've recently installed Ruby you may run into this issue. To fix it, search your Rakefile for ".exists" and replace it with ".exist"

This is such a fun issue!! I'm so glad I spent an entire day trying to figure it out!!

I hope I may save someone some time by putting this out there. Please respond here or email us at collectionbuilder.team@gmail.com with any questions. 

yrs, @dcnb

**Comment by dcnb** [comment]
this edit had already been made

**Comment by evanwill** [comment]
All, just to be clear about who this issue might apply to--> `.exists?` method was used in the CB-CSV Rakefile until April 2023 when it was updated to `.exist?`. If you are using a CB template from before spring 2023 *and* have more recently updated your Ruby version, you might encounter this error (`.exists?` was removed in Ruby 3.2.0+). To fix the error, you probably want to grab a the updated version of the [Rakefile from the current CB-CSV repository](https://github.com/CollectionBuilder/collectionbuilder-csv/blob/main/Rakefile) and replace the old one in your repository since the task was [completely rewritten](https://collectionbuilder.github.io/cb-docs/docs/changes/#rake-tasks-updated)!
